### PR TITLE
bump node version to 18

### LINKED ISSUE #316: Upgrade the docs Vercel project to NodeJS v18
Vercel will end support for NodeJS v16 on August 15, 2023 (https://vercel.com/changelog/node-js-14-and-16-are-being-deprecated). Let's make sure our docs site is ready for this!

### PR DESCRIPTION
**Pull request by benarent**
Closes #316

Bump Node to v18, tested locally and no issues. Will also fix the storybook test that needs node 18

### CODE REVIEW

File: content/14.x
Comment (ptgott) [inline content/14.x:1]: Is this change intentional?

Comment (benarent) [reply content/14.x:1]: no, that was an accident. Just cleaned it up and did a fresh sub-module update.

Comment (ptgott) [reply content/14.x:1]: 👍 There shouldn't be any need to bump submodules, since Vercel does a fresh `update-and-build` before building the docs site. That said, it doesn't _hurt_ anything to update them, but the submodules will go out of sync the next time we merge something into a `teleport` branch.

**Review by ptgott — APPROVED** [review]
